README.md

```diff
@@ -38,9 +38,15 @@ cargo build --package=examples --examples --features=linux --release
 After compilation, the modules can be found in the path `target/release/examples/` ( with the `.so` file extension for
 Linux or `.dylib` for MacOS).
 
-Additionally, the folder  `.cache/nginx/{NGX_VERSION}/{OS}/` will contain the compiled version of NGINX used to build
+Additionally, the folder  `.cache/nginx/{NGX_VERSION}/{OS}-{Arch}/` will contain the compiled version of NGINX used to build
```

Comment:
If we are going to encode architecture here, we may want to consider over to using the full [cargo triple string](https://doc.rust-lang.org/cargo/appendix/glossary.html#target) because os and architecture are not enough information to tell you if you can actually run the binary on a system. For example, a library may be build for `musl`, and will not work on other glibc variants.

It looks like `<arch><sub>-<vendor>-<sys>-<abi>`. 

What do you think of doing that instead?

Reply:
Cargo triple strings contain enough information to replace `{OS}-{Arch}` since we currently only support linux and macos (if not, there are more variants listed [here](https://doc.rust-lang.org/std/env/consts/constant.OS.html) and [here](https://doc.rust-lang.org/std/env/consts/constant.ARCH.html)). What I suspect is whether the triple is enough. Does Nginx binary vary depending on Linux distributions? For example, do we have to distinguish two same version Nginxs, one of which is compiled on Debian and the other is on CentOS?

Reply:
> Does Nginx binary vary depending on Linux distributions

It does only for the version created by the distros for their packaging systems. Many of those contain patches. Those binaries are considered "unsupported" / "unsupportable" from the perspective of NGINX. However, that doesn't really matter for ngx-rust, because we are building the binaries from scratch without any such patches, so I believe we do not need to encode any distro specific information. In short, we don't need to care about this for ngx-rust.

Let's change the PR to support the triple string, and I'll click the merge button.